File: modules/mediation/android/src/main/java/games/tateyama/godot/services/mediation/MediationService.java
package games.tateyama.godot.services.mediation;

import com.google.android.gms.ads.MobileAds;
import android.content.Context;

class MediationService {
	private final Context context;
	private final long nativeHandle;

	public MediationService(Context context, long nativeHandle) {
		this.context = context;
		this.nativeHandle = nativeHandle;
	}

	public void initialize() {
		MobileAds.initialize(context, (status) -> {
			nativeInitialized(nativeHandle);
		});
	}

	private native static void nativeInitialized(long nativeHandle);
}

File: platform/android/java/lib/src/games/tateyama/godot/services/mediation/MediationService.kt
package games.tateyama.godot.services.mediation
import android.content.Context
import com.google.android.gms.ads.MobileAds
import com.google.android.gms.ads.initialization.InitializationStatus
import com.google.android.gms.ads.initialization.OnInitializationCompleteListener

class MediationService(
	private val context: Context
) {
	fun initialize() {
		MobileAds.initialize(
			context,
			{ initializationStatus: InitializationStatus -> }
		)
	}

	companion object {
		@JvmStatic
		external fun onInitializationComplete(addr: Long)
	}
}
